From 59a9ac313a2b1375b6ec9e5d4a196c757135d35c Mon Sep 17 00:00:00 2001
From: Yilliee <m.taha6f@gmail.com>
Date: Sun, 25 Apr 2021 17:30:54 +0000
Subject: [PATCH 36/39] Revert "On Samsung R vendor, there is still the
 ISehMiscPower HIDL HAL, but there is also the AOSP's AIDL HAL. It was
 previously assumed this wouldn't happen, fix this assumption."

This reverts commit 526378a6e03cfca01e1fd977df181db128320c43.
---
 ...om_android_server_power_PowerManagerService.cpp | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/services/core/jni/com_android_server_power_PowerManagerService.cpp b/services/core/jni/com_android_server_power_PowerManagerService.cpp
index 0d8534458a8..7b42c04fb53 100644
--- a/services/core/jni/com_android_server_power_PowerManagerService.cpp
+++ b/services/core/jni/com_android_server_power_PowerManagerService.cpp
@@ -117,14 +117,6 @@ static HalVersion connectPowerHalLocked() {
     if (!gPowerHalHidlExists && !gPowerHalAidlExists) {
         return HalVersion::NONE;
     }
-    if (
-                   gPowerHalAidl_ == nullptr &&
-                   gPowerHalHidlV1_0_ == nullptr) {
-       ALOGE("Trying to connect to Samsung Power HAL");
-        gSehMiscPower = ISehMiscPower::getService();
-        gPowerHalHidlV1_0_ = IPowerV1_0::getService("miscpower");
-       ALOGE("Got miscpower = %d, SehMiscPower = %d", gPowerHalHidlV1_0_ != nullptr ? 1 : 0, gSehMiscPower ? 1 : 0);
-    }
     if (gPowerHalAidlExists) {
         if (!gPowerHalAidl_) {
             gPowerHalAidl_ = waitForVintfService<IPowerAidl>();
@@ -136,6 +128,12 @@ static HalVersion connectPowerHalLocked() {
             gPowerHalAidlExists = false;
         }
     }
+    if (gPowerHalHidlExists && gPowerHalHidlV1_0_ == nullptr) {
+       ALOGE("Trying to connect to Samsung Power HAL");
+        gSehMiscPower = ISehMiscPower::getService();
+        gPowerHalHidlV1_0_ = IPowerV1_0::getService("miscpower");
+        ALOGE("Got miscpower = %d, SehMiscPower = %d", gPowerHalHidlV1_0_ != nullptr ? 1 : 0, gSehMiscPower ? 1 : 0);
+    }
     if (gPowerHalHidlExists && gPowerHalHidlV1_0_ == nullptr) {
         gPowerHalHidlV1_0_ = IPowerV1_0::getService();
         if (gPowerHalHidlV1_0_) {
-- 
2.17.1

